File: allinone-kotlin/app/src/main/java/com/example/allinone/data/Investment.kt
package com.example.allinone.data

import java.util.Date

data class Investment(
    val id: Long = 0,
    val name: String,
    val amount: Double,
    val type: String,
    val description: String?,
    val imageUri: String?,
    val date: Date,
    val isPast: Boolean = false,
    val profitLoss: Double = 0.0, // Tracks profit/loss without affecting transaction balance
    val currentValue: Double = 0.0 // Current market value of the investment
) 

File: src/features/transactions/types/Investment.ts
export interface Investment {
  id: string;
  name: string;
  amount: number;
  type: string;
  description?: string;
  imageUri?: string;
  imageUris?: string; // comma-separated, same pattern as notes
  videoUris?: string; // comma-separated
  voiceNoteUris?: string; // comma-separated
  date: string; // ISO string
  isPast?: boolean;
  profitLoss?: number;
  currentValue?: number;
}
